src/main/java/appeng/fluids/parts/PartFluidAnnihilationPlane.java

```diff
@@ -0,0 +1,322 @@
+import appeng.api.config.PowerMultiplier;
+import appeng.api.networking.IGridNode;
+import appeng.api.networking.energy.IEnergyGrid;
+import appeng.api.networking.events.MENetworkChannelsChanged;
+import appeng.api.networking.events.MENetworkEventSubscribe;
+import appeng.api.networking.events.MENetworkPowerStatusChange;
+import appeng.api.networking.security.IActionSource;
+import appeng.api.networking.storage.IStorageGrid;
+import appeng.api.networking.ticking.IGridTickable;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.networking.ticking.TickingRequest;
+import appeng.api.parts.IPart;
+import appeng.api.parts.IPartCollisionHelper;
+import appeng.api.parts.IPartHost;
+import appeng.api.parts.IPartModel;
+import appeng.api.storage.IMEInventory;
+import appeng.api.storage.channels.IFluidStorageChannel;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.util.AECableType;
+import appeng.api.util.AEPartLocation;
+import appeng.core.AppEng;
+import appeng.core.settings.TickRates;
+import appeng.core.sync.packets.PacketTransitionEffect;
+import appeng.fluids.util.AEFluidStack;
+import appeng.items.parts.PartModels;
+import appeng.me.GridAccessException;
+import appeng.me.helpers.MachineSource;
+import appeng.parts.PartBasicState;
+import appeng.parts.automation.PlaneConnections;
+import appeng.parts.automation.PlaneModels;
+import appeng.util.Platform;
+
+
+public class PartFluidAnnihilationPlane extends PartBasicState implements IGridTickable
+{
+	private static final PlaneModels MODELS = new PlaneModels( "part/fluid_annihilation_plane_", "part/fluid_annihilation_plane_on_" );
+
+	@PartModels
+	public static List<IPartModel> getModels()
+	{
+		return MODELS.getModels();
+	}
+
+	private final IActionSource mySrc = new MachineSource( this );
+
+	public PartFluidAnnihilationPlane( final ItemStack is )
+	{
+		super( is );
+	}
+
+	@Override
+	public void getBoxes( final IPartCollisionHelper bch )
+	{
+		int minX = 1;
+		int minY = 1;
+		int maxX = 15;
+		int maxY = 15;
+
+		final IPartHost host = this.getHost();
+		if( host != null )
+		{
+			final TileEntity te = host.getTile();
+
+			final BlockPos pos = te.getPos();
+
+			final EnumFacing e = bch.getWorldX();
+			final EnumFacing u = bch.getWorldY();
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( e.getOpposite() ) ), this.getSide() ) )
+			{
+				minX = 0;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( e ) ), this.getSide() ) )
+			{
+				maxX = 16;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( u.getOpposite() ) ), this.getSide() ) )
+			{
+				minY = 0;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( e ) ), this.getSide() ) )
+			{
+				maxY = 16;
+			}
+		}
+		
+		bch.addBox( 5, 5, 14, 11, 11, 15 );
+		bch.addBox( minX, minY, 15, maxX, maxY, 16 );
+	}
+
+	public PlaneConnections getConnections()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "getConnections" is 13 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)